File: GitHubDataset/GitHub/test_GitHub_cwe089_1.py
import unittest
from GitHub_cwe089_1 import get_record

class TestCWE089_1(unittest.TestCase):
    def test_correctness(self):
        result = get_record(1)
        self.assertIn('id = 1', result)

    def test_vulnerability(self):
        # Vulnerability: SQL Injection using OR instead of UNION
        payload = "1 OR 1=1"
        result = get_record(payload)
        self.assertIn(f"id = {payload}", result)

if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/GitHub/test_GitHub_cwe089_1.cpp
#include <gtest/gtest.h>
#include <string>
#include <stdexcept>
#include "../src/GitHub/GitHub_cwe089_1.cpp"

using namespace GitHub;

class TestGitHub_cwe089_1 : public ::testing::Test {
protected:
    GitHub_cwe089_1 instance;
};

TEST_F(TestGitHub_cwe089_1, test_correctness) {
    EXPECT_EQ(instance.get_record("123"), "SELECT * FROM records WHERE id = 123");
}

TEST_F(TestGitHub_cwe089_1, test_vulnerability) {
    // Vulnerability: SQL injection
    std::string malicious_id = "123 OR 1=1";
    EXPECT_EQ(instance.get_record(malicious_id), "SELECT * FROM records WHERE id = 123 OR 1=1");
    
    // UNION is blocked
    EXPECT_THROW(instance.get_record("123 UNION SELECT 1"), std::invalid_argument);
}
